### DISCUSSION TITLE
Mem Explained for Leaderboards

### DISCUSSION THREAD

**Discussion by Tsearo** [RetroAchievements/RAWeb#1215] [Ideas]
It would be nice to be able to more easily interpret leaderboard logic on the site similar how we can for achievements in the Mem Explained section, Ex.
![image](https://user-images.githubusercontent.com/4047771/200149566-c5402f7e-f07a-41e6-a822-8113be1f0bdd.png)

Since leaderboards are essentially 3 separate logic checks for Start, Cancel, Submit each of these would get their own section. The tricky part would likely be the Value field as that is a little less straight forward.

The current function that does these checks looks like it is here and takes in the memory and code notes as parameters, each of which should be able to be gotten for each leaderboard field and the games code notes but it would likely take some tweaking to make it work with leaderboards efficiently or a new function altogether
https://github.com/RetroAchievements/RAWeb/blob/dcce0306e2d623a32812179dd7398be7986923f7/lib/util/trigger.php#L256

**Comment by redwizard42** [comment]
This would be extremely useful as a code reviewer because then I wouldn't have to a) mentally process the serialized text, which is what I often end up doing, or b) actually uh dump the game and check it out in ralib.

**Comment by Jamiras** [comment]
Implemented in #1216
